ci/launch_cluster.sh

```diff
@@ -10,74 +10,73 @@ if ! command -v envsubst >/dev/null 2>&1; then
     echo "envsubst was not found. Please install along with gettext."
     exit 1
 fi
+if ! command -v terraform >/dev/null 2>&1; then
+    echo "terraform was not found. Please install it."
+    exit 1
+fi
 
 # Two parameters are expected: CHANNEL and VARIANT where CHANNEL is the respective PR and
 # VARIANT could be one of three custer variants: open, strict or permissive.
-if [ "$#" -ne 3 ]; then
-    echo "Expected 3 parameters: launch_cluster.sh <channel> <variant> <deployment-name>"
+if [ "$#" -ne 4 ]; then
+    echo "Expected 4 parameters: launch_cluster.sh <channel> <variant> <deployment-name> <terraform-state>"
     echo "e.g. SHAKEDOWN_SSH_KEY_FILE='test.pem' launch_cluster.sh 'testing/pull/1739' 'open' 'si-testing-open'"
     exit 1
 fi
 
 CHANNEL="$1"
 VARIANT="$2"
 DEPLOYMENT_NAME="$3"
-CONFIG_PATH="$DEPLOYMENT_NAME.yaml"
-INFO_PATH="$DEPLOYMENT_NAME.info.json"
+TERRAFORM_VARS="$DEPLOYMENT_NAME.tfvars"
+TERRAFORM_STATE="$4"
 
 if [ "$VARIANT" == "open" ]; then
   INSTALLER="https://downloads.dcos.io/dcos/${CHANNEL}/dcos_generate_config.sh"
+  TERRAFORM_MODULE="github.com/dcos/terraform-dcos/aws"
 else
   INSTALLER="https://downloads.mesosphere.com/dcos-enterprise/${CHANNEL}/dcos_generate_config.ee.sh"
+  TERRAFORM_MODULE="git@github.com:mesosphere/enterprise-terraform-dcos/aws"
 fi
 
 echo "Using: ${INSTALLER}"
 
-# Create config.yaml for dcos-launch.
-envsubst <<EOF > "$CONFIG_PATH"
----
-launch_config_version: 1
-deployment_name: $DEPLOYMENT_NAME
-installer_url: $INSTALLER
-provider: onprem
-platform: aws
-aws_region: us-west-2
-os_name: cent-os-7-dcos-prereqs
-key_helper: true
-instance_type: m4.large
-num_public_agents: 1
-num_private_agents: 3
-num_masters: 3
-dcos_config:
-    cluster_name: $DEPLOYMENT_NAME
-    resolvers:
-        - 8.8.4.4
-        - 8.8.8.8
-    dns_search: mesos
-    master_discovery: static
-    exhibitor_storage_backend: static
-    rexray_config_preset: aws
+rm -rf "terraform-plan"
+mkdir "terraform-plan"
+terraform init -from-module "$TERRAFORM_MODULE" "terraform-plan"
+
+# Create .tfvars file for terraform.
+envsubst <<EOF > "$TERRAFORM_VARS"
+dcos_cluster_name = "$DEPLOYMENT_NAME"
```

Comment:
envsubst is not necessary to use; `<<EOF` already does environment variable substitution. Go ahead and try it in your shell:

```
A=hello
cat <<EOF
A = $A
EOF